Decide/src/main/java/DD2480/decide/CMVHandler.java

```diff
@@ -149,30 +149,17 @@ private boolean licNine() {
 		if(dataPoints.length<5||parameters.cPts+parameters.dPts>dataPoints.length-3){
 			return false;
 		}
-		boolean keepSearching = (dataPoints.length>3+parameters.cPts+parameters.dPts) ? true : false;
-		//first 3 consecutive datapoints
-		Points first = dataPoints[0];
-		Points second = dataPoints[parameters.cPts+1];
-		Points third = dataPoints[parameters.cPts+parameters.dPts+2];
-		int pos = 0;
-		for(int i = 2+parameters.cPts+parameters.dPts;i < dataPoints.length;i++){
+		for(int i = 0;i < dataPoints.length-2-parameters.cPts-parameters.dPts;i++){
 
-			if((first.x==second.x&&first.y==second.y)||(third.x==second.x&&third.y==second.y)){
+			if((dataPoints[0].x==dataPoints[parameters.cPts+1].x&&dataPoints[0].y==dataPoints[parameters.cPts+1].y)||
```

Comment:
maybe we should split this if statement up a bit more. row 154 is very long.

Reply:
Sure think I know how to do this in a good way